File: sdkwork-ai-sdk-go/api/assistants.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type AssistantsApi struct {
    client *sdkhttp.Client
}

func NewAssistantsApi(client *sdkhttp.Client) *AssistantsApi {
    return &AssistantsApi{client: client}
}

// Get assistant
func (a *AssistantsApi) GetAssistant(assistant_id string) (sdktypes.AssistantResponse, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/assistants/%s", assistant_id)), nil, nil)
    if err != nil {
        var zero sdktypes.AssistantResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantResponse](raw)
}

// Update assistant
func (a *AssistantsApi) UpdateAssistant(assistant_id string, body sdktypes.AssistantCreateRequest) (sdktypes.AssistantResponse, error) {
    raw, err := a.client.Post(AiApiPath(fmt.Sprintf("/assistants/%s", assistant_id)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.AssistantResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantResponse](raw)
}

// Delete assistant
func (a *AssistantsApi) DeleteAssistant(assistant_id string) (sdktypes.AssistantDeleteResponse, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/assistants/%s", assistant_id)), nil, nil)
    if err != nil {
        var zero sdktypes.AssistantDeleteResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantDeleteResponse](raw)
}

// List assistants
func (a *AssistantsApi) ListAssistants(query map[string]interface{}) (sdktypes.AssistantListResponse, error) {
    raw, err := a.client.Get(AiApiPath("/assistants"), query, nil)
    if err != nil {
        var zero sdktypes.AssistantListResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantListResponse](raw)
}

// Create assistant
func (a *AssistantsApi) CreateAssistant(body sdktypes.AssistantCreateRequest) (sdktypes.AssistantResponse, error) {
    raw, err := a.client.Post(AiApiPath("/assistants"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.AssistantResponse
        return zero, err
    }
    return decodeResult[sdktypes.AssistantResponse](raw)
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/assistants.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import AssistantCreateRequest, AssistantDeleteResponse, AssistantListResponse, AssistantResponse

class AssistantsApi:
    """Assistants API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def get_assistant(self, assistant_id: str) -> AssistantResponse:
        """Get assistant"""
        return self._client.get(f"/ai/v3/v1/assistants/{assistant_id}")

    def update_assistant(self, assistant_id: str, body: AssistantCreateRequest) -> AssistantResponse:
        """Update assistant"""
        return self._client.post(f"/ai/v3/v1/assistants/{assistant_id}", json=body)

    def delete_assistant(self, assistant_id: str) -> AssistantDeleteResponse:
        """Delete assistant"""
        return self._client.delete(f"/ai/v3/v1/assistants/{assistant_id}")

    def list_assistants(self, params: Optional[Dict[str, Any]] = None) -> AssistantListResponse:
        """List assistants"""
        return self._client.get(f"/ai/v3/v1/assistants", params=params)

    def create_assistant(self, body: AssistantCreateRequest) -> AssistantResponse:
        """Create assistant"""
        return self._client.post(f"/ai/v3/v1/assistants", json=body)
